CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+## 0.31.4 (1 August 2023)
+
+ENHANCEMENTS:
+
+* OTEL instrumentation ([#1327](https://github.com/hashicorp/terraform-ls/pull/1327))
+* OTEL: Instrument `PreloadEmbeddedSchema` job ([#1334](https://github.com/hashicorp/terraform-ls/pull/1334))
+* jobs: Don't wait for `GetModuleDataFromRegistry` job ([#1332](https://github.com/hashicorp/terraform-ls/pull/1332))
+* Change `nvim_create_autocmd` callback to a lambda ([#1295](https://github.com/hashicorp/terraform-ls/pull/1295))
+
+BUG FIXES:
+
+* fix: Add missing context argument to a test ([#1335](https://github.com/hashicorp/terraform-ls/pull/1335))
+* Fix link in README.md ([#1324](https://github.com/hashicorp/terraform-ls/pull/1324))
```

Comment:
```suggestion
* Fix link in README.md ([#1324](https://github.com/hashicorp/terraform-ls/pull/1324))
```

From the perspective of release-to-release, we aren't really fixing any bug (i.e. the problem was never surfaced to the user) so I think we can omit this.

And the other item is documentation update, so we can move it under `INTERNAL`.